### DISCUSSION TITLE
Ok so I still can't get working directory properly

### DISCUSSION THREAD

**Discussion by neonvoidx** [LazyVim/LazyVim#1770] [Q&A] [answered]
I'm not sure how to achieve this, but when i type `nvim .` or `nvim directoryname` i want that directory to be considered the root or even the cwd for the remainder of the time that nvim is open unless i manually type `:cd otherdirectory` inside of nvim. Is this possible? 

I really dislike that NeoTree, telescope etc are all scoping based on what buffer im on. 

My use case, I have a folder with 3 subfolders  (each subfolder is a git repo), I want to be able to grep/telescope search/neotree across all three repos, not just the current root folder  of whatever buffer tab i have open.

**Comment by folke** [comment, marked as answer]
add `vim.g.root_spec = { "cwd" }` to your `opiotns.lua`

**Reply by neonvoidx** [reply]
https://github.com/LazyVim/LazyVim/assets/25580051/e7a07cf0-00be-499d-a430-21b7fbc4955e

**Reply by neonvoidx** [reply]
I added that but it seems like when i type `nvim dirb` while in `dira` it opens initially in `dirb` then everything swaps over to `dira`

**Reply by neonvoidx** [reply]
on another note, if i `cd dirA` and dirA contains: `dirB, dirC`. If I then run `nvim` and open a file in dirC, the cwd changes to dirC

**Reply by neonvoidx** [reply]
https://github.com/jacobrreed/dotfiles/tree/master/.config/nvim

**Reply by folke** [reply]
Opening a folder in NEovim does not change the cwd. Is that what you mean?
Neovim is not supposed to change the cwd, and neither does LazyVim.

Doing `nvim foo` will start NEovim in your cwd, and will open a directory listing of `foo`. Your cwd remains unchanged

**Comment by neonvoidx** [comment]
> Opening a folder in NEovim does not change the cwd. Is that what you mean? Neovim is not supposed to change the cwd, and neither does LazyVim.
> 
> Doing `nvim foo` will start NEovim in your cwd, and will open a directory listing of `foo`. Your cwd remains unchanged

Ah ok interesting.  yes that's what I mean. Ok so thats just user A. ~~However the problem still remains if I first cd into a directory, open that directory, then run nvim, if i open a buffer in a subdirectory, the cwd changes again~~ <-- appears that this was because of `workspaces.nvim` plugin i was using

**Reply by folke** [reply]
LazyVim never changes the cwd.
You can check with `:pwd`.
If you use `:cd` inside NEovim, that will of course change your cwd.

If you don't want the auomatic root detection, that is very useful in monorepos, you can disable that with the setting I mentioned above. Root detection never changed the cwd, it only passes the root dir to some keymaps for neo-tree, telescope, terminal and lazygit. For every keymap, you can always use the uppercase version to force cwd, and dont use root.

* `<leader>ff>`: root
* `<leader>fF>`: always cwd

But to re-iterate, LazyVim **never** changes your actual working directory

**Comment by neonvoidx** [comment]
Thanks @folke ! You've made swapping to vim very easy and manageable
